wemake_python_styleguide/visitors/ast/complexity/function.py

```diff
@@ -39,6 +40,19 @@
 ]
 
 
+@final
+class _ComplexityExitMetrics(object):
```

Comment:
Let's make it a `dataclass` 
We use `attrs`

Reply:
@sobolevn One question, do dataclasses raise the `TooManyPublicAttributes` violation?

Reply:
This is a good question. They should not. Do they?

Reply:
No, they don't, it seems. However, I'm having some trouble doing the refactoring :/ I made the `@final
@attr.dataclass
class _ComplexityMetrics(object):`
and defined its attributes in this `asserts: _FuncCount = attr.ib(default=defaultdict(int))` way. Then, in the `__init__` method of the `_ComplexityCounter` class I wrote `self.metr = _ComplexityMetrics()`. When I required an attribute from the dataclass I wrote `self.metr.asserts`, for example. However, the counters aren't working right, because I probably misunderstand the way attr dataclasses work. I'm sorry for the silly problem. Can you please help me with this?

Reply:
Try `default=lambda: defaultdict(int)`

Reply:
Now when running the tests I get many errors that look like this:

`122, in _check_sub_node
    counter[node] += 1
TypeError: 'function' object is not subscriptable`

and

`199, in _check_function_signature
    for node, count in counter.items():
AttributeError: 'function' object has no attribute 'items'`

Reply:
Can you please show me the code? A gist maybe? Or push it right here.

Reply:
Here's the relevant file:
https://pastebin.com/Y1L5iAkP
As I said though, I'm probably misunderstanding the way attr dataclasses work.

Reply:
@iZafiro sorry, this was my mistake. You need to use `factory=`, not `default=`
See: http://www.attrs.org/en/stable/api.html

Reply:
@sobolevn I've created the _ComplexityMetrics dataclass now :) it cointains all simple counters for function internals. I hope this solves the two remaining requested changes?